### ISSUE TITLE
Add rule to flag when a documented argument does not exist

### ISSUE THREAD

**Issue by rhshadrach-8451** [astral-sh/ruff#16875]
### Summary

Apologies in advance if this is already somewhere on the issue tracker, I searched for quite a bit but could not find it.

When there is a documented argument that does not exist, from what I can tell it doesn't violate and of the docstring rules.

```python
def foo() -> None:
    """Do foo.

    Args:
        bar: Bar.
    """
    pass
```

It would be helpful if there was a rule to detect this, especially when refactoring.

**Comment by rhshadrach-8451** [comment]
Indeed, thanks!
